airflow/providers/databricks/sensors/table_changes.py

```diff
@@ -0,0 +1,152 @@
+    """Sensor to detect changes in a Delta table.
+
+
+    :param databricks_conn_id: Reference to :ref:`Databricks
+        connection id<howto/connection:databricks>` (templated), defaults to
+        DatabricksSqlHook.default_conn_name
+    :param http_path: Optional string specifying HTTP path of Databricks SQL Endpoint or cluster.
+        If not specified, it should be either specified in the Databricks connection's
+        extra parameters, or ``sql_endpoint_name`` must be specified.
+    :param sql_endpoint_name: Optional name of Databricks SQL Endpoint. If not specified, ``http_path`` must
+        be provided as described above, defaults to None
+    :param session_configuration: An optional dictionary of Spark session parameters. If not specified,
+        it could be specified in the Databricks connection's extra parameters., defaults to None
+    :param http_headers: An optional list of (k, v) pairs
+        that will be set as HTTP headers on every request. (templated).
+    :param catalog: An optional initial catalog to use.
+        Requires DBR version 9.0+ (templated), defaults to ""
+    :param schema: An optional initial schema to use.
+        Requires DBR version 9.0+ (templated), defaults to "default"
+    :param table_name: Table name to generate the SQL query, defaults to ""
+    :param handler: Handler for DbApiHook.run() to return results, defaults to fetch_all_handler
+    :param client_parameters: Additional parameters internal to Databricks SQL Connector parameters.
+    :param timestamp: Timestamp to check event history for a Delta table.
+    :param change_filter_operator: Operator to specify filter condition to check table changes,
+        defaults to >=.
+    """
+
+    template_fields: Sequence[str] = ("databricks_conn_id", "catalog", "schema", "table_name")
+
+    def __init__(
+        self,
+        table_name: str,
+        timestamp: datetime | None = None,
+        change_filter_operator: str = ">=",
+        *args,
+        **kwargs,
+    ) -> None:
+        super().__init__(*args, **kwargs)
+        self.timestamp = timestamp
+        self.caller = "DatabricksTableChangesSensor"
+        self.change_filter_operator = change_filter_operator
+        self.table_name = table_name
+
+    def _sql_sensor(self, sql):
+        hook = self._get_hook()
+        sql_result = hook.run(
+            sql,
+            handler=self.handler if self.do_xcom_push else None,
+        )
+        return sql_result
+
+    def _generate_query(
+        self,
+        prefix: str,
+        suffix: str,
+        table_name: str,
+        time_range_filter: str,
+    ) -> str:
+        formatted_opts = f"{prefix} {table_name}{suffix}{time_range_filter}"
+        self.log.debug("Formatted options: %s", formatted_opts)
+
+        return formatted_opts.strip()
+
+    @staticmethod
+    def get_previous_version(context: Context, lookup_key):
+        return context["ti"].xcom_pull(key=lookup_key, include_prior_dates=True)
```

Comment:
We store metadata on the most recently queried version for that table. And we send/receive them using `xcom`. On querying the most recent version from Databricks, we compare it with the one stored in the metadata and take an action accordingly.

Reply:
I don't believe there is a guarantee that the most recent XCom will be pulled here. Behind the scenes `XCom.get_many()` is called and just retrieves the first record. At the mercy of the metadatabase being used there.

Also, what happens in a mapped operator situation? If the XCom key is always the same, it seems possible this can pull an XCom key for an entirely different task since `task_ids` and `map_index` is not specified in the `xcom_pull()` call.

Another question then would be what if the input args are the same (i.e. checking for changes in the same table) but a user simply updates the `task_id`. Would this sensor yield a false positive that there was indeed a change?

I don't necessarily have answers to these questions on the top of my head, but some things to think about with using XComs in this way.